providers-sdk/v1/inventory/inventory.go

```diff
@@ -318,13 +320,20 @@ var (
 	FAMILY_WINDOWS = "windows"
 )
 
-func (p *Platform) IsFamily(family string) bool {
-	for i := range p.Family {
-		if p.Family[i] == family {
-			return true
+// Merge performs a deep merge of the provided platform.
+func (p *Platform) Merge(pf *Platform) {
+	if pf != nil {
+		if err := mergo.Merge(p, pf, mergo.WithOverride); err != nil {
```

Comment:
How do we ensure it does the right merge? Do we handle precedence?

Reply:
Yes we do. There are two scenarios.
1. Default - When the new platform struct HAS values and the origin does NOT have, we set them
2. WithOverride - When the new platform struct HAS values, we set it

For both options, there has to bee non-zero values to take action.